### PR TITLE
Work around the hang issue that happens when running class basic parsing tests

### PR DESCRIPTION
**Pull request by daxian-dbw**
CrossGen'ed assemblies cause a hang to happen intermittently when running powershell class basic parsing tests in Linux/OSX. The hang seems to happen when generating dynamic assemblies.
This issue has been reported to CoreCLR team. We need to work around it for now because the Travis CI build failures caused by this is draining our builder resource and severely affect our daily work. The workaround is:
  1. For pull request and push commit, build without '-CrossGen' and run the parsing tests
  2. For nightly build, build with '-CrossGen' but don't run the parsing tests

With this workaround, CI builds triggered by pull request and push commit will exercise the parsing tests with IL assemblies, while nightly builds will exercise CrossGen'ed assemblies without running those class parsing tests so as to avoid the hang.
NOTE: this change should be reverted once the 'CrossGen' issue is fixed by CoreCLR. The CoreCLR issue is https://github.com/dotnet/coreclr/issues/9745.

### CODE REVIEW

File: tools/travis.ps1 [hunk 1]
```diff
@@ -96,7 +96,19 @@ Write-Host -Foreground Green "Executing travis.ps1 `$isPR='$isPr' `$isFullBuild=
...
+# basic parsing tests in Linux/OSX. The hang seems to happen when generating dynamic assemblies.
+# This issue has been reported to CoreCLR team. We need to work around it for now because
+# the Travis CI build failures caused by this is draining our builder resource and severely
+# affect our daily work. The workaround is:
+#  1. For pull request and push commit, build without '-CrossGen' and run the parsing tests
+#  2. For nightly build, build with '-CrossGen' but don't run the parsing tests
+# With this workaround, CI builds triggered by pull request and push commit will exercise
+# the parsing tests with IL assemblies, while nightly builds will exercise CrossGen'ed assemblies
+# without running those class parsing tests so as to avoid the hang.
+# NOTE: this change should be reverted once the 'CrossGen' issue is fixed by CoreCLR.
```
Comment (TravisEz13) [inline tools/travis.ps1]: Can we include a link to either our issue tracking this or CoreCLR's issue tracking this issue.  So someone reading the code can easily know if they resource revert this code?

Comment (daxian-dbw) [inline tools/travis.ps1]: Good point. I will add the link to the CoreCLR issue once I open it.

Comment (iSazonov) [inline tools/travis.ps1]: Maybe not remove `Start-PSBuild -CrossGen -PSModuleRestore ` and merge with comments?

**Review by lzybkr — APPROVED** [review]

File: tools/travis.ps1 (hunk 1 above)
Comment (TravisEz13) [inline tools/travis.ps1]: `Start-PSBuild` wasn't removed, just moved to line 111

Comment (iSazonov) [inline tools/travis.ps1]: O! I see now. Thanks!

**Review by mirichmo — APPROVED** [review]

**Comment by daxian-dbw** [comment]
@TravisEz13 the CoreCLR issue link is added to the comment. Please take another look.

**Pull request merged**